osu.Game/Screens/Edit/Compose/Components/SelectionBoxRotationHandle.cs

```diff
@@ -113,9 +113,13 @@ protected override void OnDragEnd(DragEndEvent e)
 
         private float convertDragEventToAngleOfRotation(DragEvent e)
         {
-            // Adjust coordinate system to the center of SelectionBox
-            float startAngle = MathF.Atan2(e.LastMousePosition.Y - selectionBox.DrawHeight / 2, e.LastMousePosition.X - selectionBox.DrawWidth / 2);
-            float endAngle = MathF.Atan2(e.MousePosition.Y - selectionBox.DrawHeight / 2, e.MousePosition.X - selectionBox.DrawWidth / 2);
+            // Adjust coordinate system to the center of the selection
+            Vector2 center = rotationHandler?.DefaultOrigin is not null
+                ? selectionBox.ToLocalSpace(rotationHandler.ToScreenSpace(rotationHandler.DefaultOrigin.Value))
+                : selectionBox.DrawSize / 2;
```

Comment:
Okay I guess that's my question above answered, but I'm not happy with this result either - can this just receive _the actual origin of the rotation that is being performed_ rather than play the assumption game about default origin being null or not?

As in I don't want to see `DefaultOrigin` here. I want to see the actual `Origin` that something higher up has deemed appropriate. In particular that might allow implementing movable origin on the selection box in the future which is something I wouldn't mind seeing.

Reply:
Right now the selection box uses the default origin, so that is the actual origin of the rotation that is being performed. I don't really understand what you're asking of me here. There is no higher up that controls the origin of the selection box. If you want to implement a movable origin on the selection box you can always provide a custom origin to the rotation handler, but that's not what I'm doing in this PR.

Reply:
Then let me ask this: why is this doing things depending on whether default origin is or isn't null?

Reply:
I'll just make it assert for not null. That way it'll throw an exception if the origin is missing.